### DISCUSSION TITLE
hideFromIndex still executed on index view

### DISCUSSION THREAD

**Discussion by joostvanhoof** [laravel/nova-issues#3439] [Q&A] [answered]
- Laravel Version: 8.46.0
- Nova Version: 3.26.1
- PHP Version: 8.0.5

### Description:
I have this code in a Resource:
```php
// Price on index page
Currency::make('Price')
    ->sortable()
    ->currency('EUR')
    ->onlyOnIndex(),

// Price detail page
Currency::make('Price')
    ->rules('required', 'numeric', 'nullable')
    ->nullValues(['', 'null', null])
    ->currency('EUR')
    ->help(view('nova.product.price', ['price_info' => $this->getPriceInfo()])->render())
    ->hideFromIndex(),
```

The problem is that the second 'Price' seems to be executed on the index page while it has `hideFromIndex()`. Not sure if this is a bug or expected behaviour.

I found out because `getPriceInfo()` is a function that has 3 queries to get some price information from the database (eg an average based on all records, so something that's only needed in the detail and edit views). With 100 products shown on the index page that means 300 queries.

Is there a way to make sure that this is only executed on the detail and edit views?

Please let me know if you need additional information.

**Comment by crynobone** [comment, marked as answer]
Yes, that is an expected behavior, all `fields` need to pull into a collection and then only filter based on display/hidden for each view type. 

However, you may configure different `fieldForIndex()` and `fieldForDetail()` to explicitly change this behavior.

**Reply by vFire** [reply]
Thanks, but I don't know how to do set as fieldForIndex() or fieldForDetail(), is it already there?
It is quite important in real business project, as the records increased, especially with flexible content related field, it will be very slow because it is still execute even not show on index page.
